__tests__/components/TableRow.spec.ts

```diff
@@ -0,0 +1,370 @@
+                targetAddress: 'TCABUWAK5WMJ26ZPERMGWBOWAJF4XPNCJOWPAAI',
+                targetId: '7E77578D00C26DFC',
+                value: 'metavalue',
+            },
+        ],
+        name: 'helloworld',
+    };
+
+    const getTableRowWrapper = (prop = {}) => {
+        return getComponent(TableRow, {}, {}, prop, {
+            Poptip: true,
+            Icon: true,
+        });
+    };
+
+    describe('isNamespace', () => {
+        const runBasicNamespaceTests = (rowValues, expectedResult) => {
+            // Arrange:
+            const wrapper = getTableRowWrapper({
+                rowValues,
+            });
+
+            const vm = wrapper.vm as TableRowTs;
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(vm.isNamespace).toBe(expectedResult);
+            // @ts-ignore
+            expect(vm.isRootNamespace).toBe(expectedResult);
+        };
+
+        test('returns true when root namespace provided', () => {
+            runBasicNamespaceTests(mockNamespaceRowValue, true);
+        });
+
+        test('returns false when not namespace', () => {
+            runBasicNamespaceTests(mockMosaicValue, false);
+        });
+
+        test('returns false when sub namespace provided', () => {
+            // Arrange:
+            const wrapper = getTableRowWrapper({
+                rowValues: {
+                    ...mockNamespaceRowValue,
+                    name: 'hello.world',
+                },
+            });
+
+            const vm = wrapper.vm as TableRowTs;
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(vm.isNamespace).toBe(true);
+            // @ts-ignore
+            expect(vm.isRootNamespace).toBe(false);
+        });
+    });
+
+    describe('hasAvailableActions', () => {
+        const runBasicHasAvailableActionsTests = (props, expectedResult) => {
+            // Arrange:
+            const wrapper = getTableRowWrapper({
+                assetType: 'mosaic',
+                ...props,
+            });
+
+            const vm = wrapper.vm as TableRowTs;
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(vm.hasAvailableActions).toBe(expectedResult);
+            expect(wrapper.find('.pop-tip').exists()).toBe(expectedResult);
+
+            if (expectedResult) {
+                wrapper.find('.poptip-actions').trigger('click');
+                expect(wrapper.emitted('on-show-alias-form')).toBeDefined();
+            } else {
+                expect(wrapper.find('.poptip-actions').exists()).toBe(false);
+            }
+        };
+
+        test('hide Poptip component when assert type is account restrictions', () => {
+            runBasicHasAvailableActionsTests(
+                {
+                    assetType: 'accountRestrictions',
+                    rowValues: {
+                        ...mockNamespaceRowValue,
+                    },
+                },
+                false,
```

Comment:
why actions hidden for account restrictions ?

Reply:
Because TableRow is used to display mosaics and namespace in the table. `AccountRestrictions` is not supported.

<img width="476" alt="image" src="https://user-images.githubusercontent.com/5649156/186227633-5a2fcd74-9208-4cc1-b156-c4ac8673b2dc.png">